src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -1735,18 +1735,35 @@ address TemplateInterpreterGenerator::generate_trace_code(TosState state) {
 }
 
 void TemplateInterpreterGenerator::count_bytecode() {
-  __ push_reg(t0);
-  __ push_reg(x10);
-  __ mv(x10, (address) &BytecodeCounter::_counter_value);
-  __ mv(t0, 1);
-  __ amoadd_d(zr, x10, t0, Assembler::aqrl);
-  __ pop_reg(x10);
-  __ pop_reg(t0);
+  __ mv(x7, (address) &BytecodeCounter::_counter_value);
+  __ atomic_addalw(noreg, 1, x7);
```

Comment:
I think a simpler __ atomic_addw will do here? I don't think there is any memory ordering issue here.

Reply:
Thank you for your review. Fixed.